mongo/integration/primary_stepdown_test.go

```diff
@@ -23,7 +24,65 @@ const (
 	errorInterruptedAtShutdown int32 = 11600
 )
 
+// testPoolMonitor exposes an *event.PoolMonitor and collects all events logged to that
+// *event.PoolMonitor. It is safe to use from multiple concurrent goroutines.
```

Comment:
Oh very nice. Optional: should this be in a separate file, like `connection_pool_helpers_test.go`? Though `isPoolCleared` was used by other files, the name of this file suggests it is only for the tests specified in https://github.com/mongodb/specifications/blob/master/source/connections-survive-step-down/tests/README.rst

Reply:
Yeah, I wanted to locate it "next to" the existing `poolMonitor` so that there aren't two implementations of the same thing in different places. I think the `testPoolMonitor` could actually be a good candidate for a "test utilities" package because numerous test packages need to record events (including the `Server` tests added in this PR). However, I'm not sure it's worth moving into a separate file until we want to move it into a test utilities package, which probably shouldn't be part of this PR.